os-fpga/FOEDAG
# [RG-93] Project settings

### Motivate of the pull request
Allow user modify project settings

### Describe the technical details
When project created user can modify project settings:
* via context menu in the project hierarchy
* via menu Project->Project setting

 ### Which part of the code base require a change
 
 - [x] Library: newproject, fordagcore, projnavigator
 - [ ] Plug-in: <Specify the plugin name>
 - [ ] Continous Integration (CI) scripts

### Testcases
#### Testcase1:
* run any script from testcase folder.
* copy somewhere project file
* Open and close project settings
* Compare latest project file with the one that was copied.

#### Testcase2:
* Create project with all available options
* Open project settings.
* It is expected that project settings are the same as it was created.

#### Testcase3:
* create project with some settings
* Open project settings and close several times.
* Nothing should be changed

#### Testcase4:
* create project with some settings
* Open project settings
* Change something
* Close settings and open again
* All changes should be saved.
* Repeat steps with all possible settings


### Screenshots
![image](https://user-images.githubusercontent.com/95262932/193937255-69e468a0-9c1b-45f9-9e71-f2198d05fefa.png)
![image](https://user-images.githubusercontent.com/95262932/193937407-69160adc-3bf6-42e5-86ee-76af66e95353.png)
![image](https://user-images.githubusercontent.com/95262932/193937446-d49c3f81-bba1-47ba-8f37-0346f5b3647f.png)
